finalProject/tenSpeckledFrogs.js

```diff
@@ -0,0 +1,143 @@
+      rhyme += currMsg;
+    } else if (i === numInitialFrogs) { // build last iteration msg
+      currMsg = `One speckled frog sat on a log eating some most delicious bugs.\nOne jumped in the pool where it's nice and cool,\nthen there were no more speckled frogs!`;
+      rhyme += currMsg;
+    } else if (currNumFrogs > 2) { // any other inner msg
+      var currNum = numToString(currNumFrogs);
+      var currNumCap = currNum.charAt(0).toUpperCase() + currNum.slice(1);
+      currMsg = `${currNumCap} speckled frogs sat on a log eating some most delicious bugs.\nOne jumped in the pool where it's nice and cool,\nthen there was ${numToString(currNumFrogs - 1)} speckled frogs.\n\n`;
+      rhyme += currMsg;
+    } else if (currNumFrogs === 2) { // if frogs is two, need specific formatting for one frog left on last line
+      currMsg = `Two speckled frogs sat on a log eating some most delicious bugs.\nOne jumped in the pool where it's nice and cool,\nthen there was one speckled frog.\n\n`;
+      rhyme += currMsg;
+    }
+    currNumFrogs -= 1;
+  }
+
+
+  // rhyme contains the entire string, not yet formatted
+  return rhyme;
+}
+
+// iterate through whole string,
+// at numbers (not NaN and not \n)
+  // also check if next element is number
+    // if it is, build a separate string and keep storing values until a value is not a number
+  // call helper function
+
+  // make sure beginnings of sentences are capitalized
+
+
+function numToString(num) {
+  var fullString = ``;
+  var ones = {
+    0 : '',
+    1 : 'one',
+    2 : 'two',
+    3 : 'three',
+    4 : 'four',
+    5 : 'five',
+    6 : 'six',
+    7 : 'seven',
+    8 : 'eight',
+    9 : 'nine'
+  };
+
+  var teens = {
+    10 : 'ten',
+    11 : 'eleven',
+    12 : 'twelve',
+    13 : 'thirteen',
+    14 : 'fourteen',
+    15 : 'fifteen',
+    16 : 'sixteen',
+    17 : 'seventeen',
+    18 : 'eighteen',
+    19 : 'nineteen'
+  };
+
+  var tens = {
+    0 : '',
+    2 : 'twenty',
+    3 : 'thirty',
+    4 : 'fourty',
+    5 : 'fifty',
+    6 : 'sixty',
+    7 : 'seventy',
+    8 : 'eighty',
+    9 : 'ninety'
+  };
+
+  var above = ['hundred', 'thousand', 'million', 'billion'];
+
+  var orig = num;
+  num = JSON.stringify(num);
+  var length = num.length;
+  if (length >= 3) {
+    var onesAndTens = num.slice(-2);
+    var rest = num.slice(0, -2);
+    console.log(rest);
+    if (teens[onesAndTens] !== undefined) {
+      var totalOnesTens = teens[onesAndTens];
+    } else {
+      var setTens = onesAndTens[0];
+      var localTens = tens[setTens];
+      var setOnes = onesAndTens[1];
+      var localOnes = ones[setOnes];
+      var totalOnesTens = `${localTens}-${localOnes}`;
+      if (localOnes === '' || localTens === '') {
+        if (localOnes === '' && localTens === '') {
+          var formatHundredth = `${ones[rest]} ${above[0]}`;
+          return formatHundredth;
+        }
+        if (localOnes === '') {
```

Comment:
what is the essence of this block and the one below. Could it be abstracted into its own function